explorer/src/utils/security-txt.ts

```diff
@@ -0,0 +1,71 @@
+import { ProgramDataAccountInfo } from "validators/accounts/upgradeable-program";
+
+export type SecurityTXT = {
+  name: string;
+  project_url: string;
+  contacts: string;
+  policy: string;
+  preferred_languages?: string;
+  source_code?: string;
+  encryption?: string;
+  auditors?: string;
+  acknowledgements?: string;
+  expiry?: string;
+};
+
+const HEADER = "=======BEGIN SECURITY.TXT V1=======\0";
+const FOOTER = "=======END SECURITY.TXT V1=======\0";
+
+export const fromProgramData = (
+  programData: ProgramDataAccountInfo
+): SecurityTXT | undefined => {
```

Comment:
After you check that the security.txt entry has all the expected keys, can you return an error if security.txt is present but some expected keys are missing or the security.txt is invalid in some way so that devs can debug issues here?

Reply:
The code now throws an error if not all required fields can be found.